ignite/handlers/checkpoint.py

```diff
@@ -295,7 +299,21 @@ def __call__(self, engine: Engine) -> None:
                 for k in checkpoint:
                     name = k
                 checkpoint = checkpoint[name]
-            filename = "{}{}_{}{}".format(self._fname_prefix, name, suffix, self._ext)
+            filename_pattern_dict["name"] = name
+
+            filename = Checkpoint.filename_pattern.format(**filename_pattern_dict)
+
+            if not self.filename_prefix:
```

Comment:
I realized that in L306-316 post processing is done using the `filename_pattern`.

For instance, If a user defines a pattern `{filename_prefix}__{name}.{ext}` without `global_step_transform`, `__` will be replace and that is not wanted here.

Reply:
yes @sdesrozis you are right, but that part was a bit ugly so changed the code to what @vfdev-5 has suggested.

I can change this back if you are ok. But when going back will introduce a new class attribute. 
https://github.com/pytorch/ignite/pull/1127/commits/1802709e86c8068c84c804b829806da03385abec#diff-18cd1a3c06de70b0e34b8fcf32940b30R203-R204

Reply:
I think that comment from @vfdev-5 was about introduction of public class members. We should have a public attribute (for user) and others should be private
```
filename_pattern  = "..." #public
_filename_pattern = filename_pattern # private
```
As you did in your code, we can compare.

Reply:
@sdesrozis I shall look into the implementation of introducing an additional argument to `__init__` as @vfdev-5 has mentioned it will cause issues when having  2 savers.